xbmc/cores/paplayer/PAPlayer.cpp

```diff
@@ -804,7 +813,9 @@ inline bool PAPlayer::ProcessStream(StreamInfo *si, double &freeBufferTime)
       UpdateStreamInfoPlayNextAtFrame(m_currentStream, m_upcomingCrossfadeMS);
 
       UpdateGUIData(si);
-      m_callback.OnPlayBackStarted(si->m_fileItem);
+      if (m_signalStarted)
+        m_callback.OnPlayBackStarted(si->m_fileItem);
+      m_signalStarted = true;
```

Comment:
As these three lines are repeated five times, i would factor them out to a new method. Even the call to `m_callback.OnAVStarted` following these three lines could be put into this new method.

Reply:
that would imply that those functions calls belong together but they don't. if done correctly, it would look differently in those five locations.

Reply:
Okay, then, forget about the onavstarted  part of my suggestion.

But I still thing that m_signalstarted member logic and calling onplaybackstarted belong together and thus encapsulating this into a separate function makes sense.

Reply:
@ksooo I factored the error case into a method

btw: considering current master this is not a fix but an improvement

Reply:
Nice. Thanks.